ci/ray_ci/automation/docker_tags_lib.py

```diff
@@ -232,3 +234,129 @@ def query_tags_from_docker_with_oci(namespace: str, repository: str) -> List[str
+
+
+def _is_release_tag(
+    tag: str,
+    release_versions: Optional[List[str]] = None,
+) -> bool:
+    """
+    Check if tag is a release tag & is in the list of release versions.
+    Tag input format should be just the tag name, without namespace/repository.
+    Tag input can be in any format queried from Docker Hub: "x.y.z-...", "a1s2d3-..."
+    Args:
+        tag: Docker tag name
+        release_versions: List of release versions.
+            If None, don't filter by release version.
+    Returns:
+        True if tag is a release tag and is in the list of release versions.
+            False otherwise.
+    """
+    versions = tag.split(".")
+    if len(versions) != 3 and "post1" not in tag:
+        return False
+    # Parse variables into major, minor, patch version
+    major, minor, patch = versions[0], versions[1], versions[2]
+    extra = versions[3] if len(versions) > 3 else None
+    if not major.isnumeric() or not minor.isnumeric():
+        return False
+    if not patch.isnumeric() and "rc" not in patch and "-" not in patch:
+        return False
+
+    if "-" in patch:
+        patch = patch.split("-")[0]
+    release_version = ".".join([major, minor, patch])
+    if extra:
+        release_version += f".{extra}"
+    if release_versions and release_version not in release_versions:
+        return False
+
+    return True
+
+
+def _crane_binary():
+    r = runfiles.Create()
+    system = platform.system()
+    if system != "Linux" or platform.processor() != "x86_64":
+        raise ValueError(f"Unsupported platform: {system}")
+    return r.Rlocation("crane_linux_x86_64/crane")
+
+
+def _call_crane_cp(tag: str, source: str, aws_ecr_repo: str):
+    try:
+        with subprocess.Popen(
+            [
+                _crane_binary(),
+                "cp",
+                source,
+                f"{aws_ecr_repo}:{tag}",
+            ],
+            stdout=subprocess.PIPE,
+            text=True,
+        ) as proc:
+            output = ""
+            for line in proc.stdout:
+                logger.info(line + "\n")
+                output += line
+            return_code = proc.wait()
+            if return_code:
+                raise subprocess.CalledProcessError(return_code, proc.args)
+            return output
+    except subprocess.CalledProcessError as e:
+        return f"Error: {e.output}"
+
+
+def copy_tag_to_aws_ecr(tag: str, aws_ecr_repo: str) -> bool:
+    """
+    Copy tag from Docker Hub to AWS ECR.
+    Args:
+        tag: Docker tag name in format "namespace/repository:tag"
+    Returns:
+        True if tag was copied successfully, False otherwise.
+    """
+    _, repo_tag = tag.split("/")
+    tag_name = repo_tag.split(":")[1]
+    logger.info(f"Copying from {tag} to {aws_ecr_repo}:{tag_name}......")
+    result = _call_crane_cp(
+        tag=tag_name,
+        source=tag,
+        aws_ecr_repo=aws_ecr_repo,
+    )
+    if "Error" in result:
```

Comment:
We are still relying on the content of the process stdout for error handlings; would it be safer to do [1] instead of relying on the presence of the word 'Error'?

Reply:
Thank you! Just fixed it!